PokeDemo/Controllers/PokemonDetailViewController.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  PokemonDetailViewController.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 05/07/22.
+//
+// MARK: - Navigation
+
+import UIKit
+
+class PokemonDetailViewController: UIViewController {
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var pokemonImageView: UIImageView!
+    
+    // MARK: - Properties
+    
+    private let pokemonDetailViewModel = PokemonDetailViewModel()
+    var pokemonName: String!
+    
+    // MARK: - View Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = pokemonName.uppercased()
+        getPokemonDetails()
+    }
+    
+    // MARK: - API methods
+    
+    private func getPokemonDetails() {
+        
+        pokemonDetailViewModel.getPokemonDetail(pokemonName: ("/" + pokemonName + "/")) {[weak self] success, error in
+            guard let strongSelf = self else { return }
+            if success ?? false {
+                let url = URL(string: strongSelf.pokemonDetailViewModel.imageURL!)
+                let data = try? Data(contentsOf: url!)
+                strongSelf.pokemonImageView.image = UIImage(data: data!)
+            } else {
+                AlertUtility.showAlert(strongSelf, title: AlertUtility.AlertTitles.error, message: error?.localizedDescription)
```

Comment:
**issue:** This is not MVVM works. Find appropriate article or example app and share with me first and then implement MVVM that way.

Reply:
I suggest you to use MVP for UIKit app instead of MVVM as UIKit doesn't provide reactive paradigm inbuilt the way swiftUI provides. Thus it bleeds out some of the logics in the view layer and you can't avoid it. But MVP provides more control and keeps logic within presenter layer and covers all the logic unit tests

Reply:
Added presenter file.

Reply:
Fixed in [ac902cf](https://github.com/rudrbansal/Pokemon/pull/1/commits/ac902cf439c49644e5048c30bcb7da757e1c78b3)